jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -110,31 +122,105 @@
+    }
 
-        this.getLog().info("Generating JSON Schema for class: " + this.className);
+    /**
+     * Generate the JSON schema for the given className.
+     *
+     * @param className The name of the class
+     * @throws MojoExecutionException In case of problems
+     */
+    private void generateSchema(String className) throws MojoExecutionException {
         // Load the class for which the schema will be generated
         Class<?> schemaClass;
         try {
-            schemaClass = Class.forName(this.className);
+            schemaClass = Class.forName(className);
         } catch (ClassNotFoundException e) {
-            throw new MojoExecutionException("Error loading class " + this.className, e);
+            throw new MojoExecutionException("Error loading class " + className, e);
         }
-        JsonNode jsonSchema = generator.generateSchema(schemaClass);
+
+        this.generateSchema(schemaClass);
+    }
+
+    /**
+     * Generate the JSON schema for the given className.
+     *
+     * @param schemaClass The class for which the schema is to be generated
+     * @throws MojoExecutionException In case of problems
+     */
+    private void generateSchema(Class<?> schemaClass) throws MojoExecutionException {
+        JsonNode jsonSchema = getGenerator().generateSchema(schemaClass);
         File file = getSchemaFile(schemaClass);
         this.getLog().info("- Writing schema to file: " + file);
         this.writeToFile(jsonSchema, file);
     }
 
+    /**
+     * Generate JSON schema's for all classes in a package.
+     *
+     * @param packageName The name of the package
+     * @throws MojoExecutionException in case of problems
+     */
+    private void generateSchemaForPackage(String packageName) throws MojoExecutionException {
+        Reflections reflections = new Reflections(packageName, new SubTypesScanner(false));
+        Set<Class<?>> subTypes = reflections.getSubTypesOf(Object.class);
+        for (Class<?> mainType : subTypes) {
+            this.generateSchema(mainType);
+        }
+    }
+
+    /**
+     * Build up the initial logging text.
+     *
+     * @return The info message text
+     */
+    private String getInfoText() {
+        StringBuilder logText = new StringBuilder("Generating JSON Schema for ");
+        if (classNames.length > 0) {
+            logText.append("class");
+            if (classNames.length > 1) {
+                logText.append("es");
+            }
+            logText.append(" ");
+            logText.append(String.join(", ", classNames));
+            if (packageNames.length > 0) {
+                logText.append(" and ");
+            }
+        }
+        if (packageNames.length > 0) {
+            logText.append("package");
+            if (packageNames.length > 1) {
+                logText.append("s");
+            }
+            logText.append(" ");
+            logText.append(String.join(", ", packageNames));
+        }
+
+        return logText.toString();
+    }
```

Comment:
This is a summary of everything that was configured. While informative, I would assume the user knows what they configured.

I'd deem this to be more helpful if a log message was written for each class/package at the time of the corresponding schema being generated. That would give some progress indication and make it more obvious in case of errors – at what specific class/package it occurred.

Reply:
This is also a matter of taste. I leave it up to you.
You could kick this code out if you like. I am fine with all solutions.